homeassistant/components/sensor/yr.py

```diff
@@ -82,8 +82,9 @@ def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
 
     weather = YrData(hass, coordinates, dev)
     # Update weather on the hour, spread seconds
-    async_track_utc_time_change(hass, weather.async_update, minute=0,
-                                second=randrange(5, 25))
+    async_track_utc_time_change(hass, weather.async_update,
+                                minute=randrange(1, 10),
```

Comment:
The weather forecast change for each hour. If we use `randrange(0, 59)`, we risk showing old information for one hour.